### PR TITLE
Update React widgets to trigger off widget-type instead of url

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
With the move to Drupal, we'll want to be more flexible in how React widgets on static pages are linked to the React code. Hardcoding urls isn't a good path forward, since we won't have much visibility into how those change. Instead, we should trigger these widgets by the existence of elements with a `data-widget-type` attribute. This should let us be more flexible in linking Drupal paragraph types to our widget code.

Related content changes: https://github.com/department-of-veterans-affairs/vagov-content/pull/266

## Testing done
Tested existing widgets locally

## Acceptance criteria
- [ ] Existing widgets still work properly

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/static-pages/createCallToActionWidget.js [hunk 1]
```diff
@@ -2,8 +2,10 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import { Provider } from 'react-redux';
 
-export default async function createCallToActionWidget(store) {
-  const widgets = Array.from(document.querySelectorAll('.cta-widget'));
+export default async function createCallToActionWidget(store, widgetType) {
+  const widgets = Array.from(
+    document.querySelectorAll(`[data-widget-type="${widgetType}"]`),
```
Comment (jbalboni) [inline src/applications/static-pages/createCallToActionWidget.js]: This change isn't really necessary, I'm doing it for consistency.

**Review by rianfowler — APPROVED** [review]

File: src/applications/static-pages/static-pages-entry.js [hunk 2]
```diff
@@ -79,59 +37,79 @@ Raven.context(
 
 createAdditionalInfoWidget();
 
-if (pensionPages.has(location.pathname)) {
+if (
+  document.querySelector(
+    `[data-widget-type="${widgetTypes.PENSION_APP_STATUS}"]`,
+  )
+) {
```
Comment (U-DON) [inline src/applications/static-pages/static-pages-entry.js]: Are these ifs necessary? It seems all of the `create*` functions already do something like:

```
const root = document.querySelector(...);
if (root) { ... }
```

**Review by U-DON — APPROVED** [review]
LGTM. Noticed something that might be redundant.

File: src/applications/static-pages/static-pages-entry.js (hunk 2 above)
Comment (U-DON) [reply src/applications/static-pages/static-pages-entry.js]: It probably isn't that expensive, but maybe we can save ourselves one DOM search for each widget.

Comment (jbalboni) [reply src/applications/static-pages/static-pages-entry.js]: Yeah, I can't see a reason why we shouldn't do that, now that we have a unique type to check.

**Review by cvalarida — APPROVED** [review]
This is definitely one of those, "Oh hey! Good idea! We should have done it this way from the beginning!" times. :stuck_out_tongue: 

Looks great!

**Pull request merged**